x/uibc/quota/ibc_module.go

```diff
@@ -41,17 +41,23 @@ func (im IBCMiddleware) OnAcknowledgementPacket(ctx sdk.Context, packet channelt
 		return errors.Wrap(err, "cannot unmarshal ICS-20 transfer packet acknowledgement")
 	}
 	if _, ok := ack.Response.(*channeltypes.Acknowledgement_Error); ok {
-		err := im.RevertQuotaUpdate(ctx, packet.Data)
-		emitOnRevertQuota(&ctx, "acknowledgement", packet.Data, err)
+		params := im.keeper.GetParams(ctx)
+		if params.IbcStatus == uibc.IBCTransferStatus_IBC_TRANSFER_STATUS_QUOTA_ENABLED {
+			err := im.RevertQuotaUpdate(ctx, packet.Data)
+			emitOnRevertQuota(&ctx, "acknowledgement", packet.Data, err)
+		}
 	}
 
 	return im.IBCModule.OnAcknowledgementPacket(ctx, packet, acknowledgement, relayer)
 }
 
 // OnTimeoutPacket implements types.Middleware
 func (im IBCMiddleware) OnTimeoutPacket(ctx sdk.Context, packet channeltypes.Packet, relayer sdk.AccAddress) error {
-	err := im.RevertQuotaUpdate(ctx, packet.Data)
-	emitOnRevertQuota(&ctx, "timeout", packet.Data, err)
+	params := im.keeper.GetParams(ctx)
+	if params.IbcStatus == uibc.IBCTransferStatus_IBC_TRANSFER_STATUS_QUOTA_ENABLED {
+		err := im.RevertQuotaUpdate(ctx, packet.Data)
+		emitOnRevertQuota(&ctx, "timeout", packet.Data, err)
```

Comment:
if there is a timeout then we should correctly revert the quota update, without checking the status.

Reply:
Why don't check the status ?

Reply:
Because we are in the timeout.